### PR TITLE
Use assertThrows in classes ActorCreationTest and OutputStreamSinkTest

### PR DESCRIPTION
**Pull request by Captain1653**
Thare are many lines look like that:
`try ... catch` in tests. It can be replaced on  [assertThrows](https://junit.org/junit4/javadoc/latest/org/junit/Assert.html#assertThrows(java.lang.Class,%20org.junit.function.ThrowingRunnable))

Kind regards

### CODE REVIEW

**Comment by johanandren** [comment]
OK TO TEST

File: akka-actor-tests/src/test/java/akka/actor/ActorCreationTest.java [hunk 1]
```diff
@@ -175,35 +176,32 @@ public void onReceive(Object message) throws Exception {}
...
-              return null;
-            }
-          });
-      assert false;
-    } catch (IllegalArgumentException e) {
-      assertEquals(
-          "cannot use non-static local Creator to create actors; make it static (e.g. local to a static method) or top-level",
-          e.getMessage());
-    }
+    IllegalArgumentException exception =
```
Comment (Captain1653) [inline akka-actor-tests/src/test/java/akka/actor/ActorCreationTest.java:179]: The test failed. I'm little surprised. I tried to save the semantics. Does  it depend on  ActorCreationTest Line 43? I have read [the documentation](https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.5.1) . It's safe to remove `static` from nested interface. Am I wrong?

**Comment by Captain1653** [comment]
Hm... It looks like some infrastructure error
`The command "curl -Ls https://git.io/jabba | bash && . ~/.jabba/jabba.sh" failed and exited with 1 during .` Am I wrong?

**Comment by johanandren** [comment]
Yeah, was something temporary yesterday, I'll trigger a rebuild.

**Comment by johanandren** [comment]
PLS BUILD

**Comment by Captain1653** [comment]
I don't understand indeed. The test `testWrongAnonymousInPlaceCreator` failed. Why? It depends on Java 8 Lambda? Should I revert my change?

**Comment by johanandren** [comment]
Unless you can figure it out and make the test pass, yes revert please.

**Comment by Captain1653** [comment]
@johanandren  review it, please

**Comment by octonato** [comment]
I think I found why that other tests was not failing. That's because the original one was also not failing: 

```java
    try {
      Props.create(
          Actor.class,
          new Creator<Actor>() {
            @Override
            public Actor create() throws Exception {
              return null;
            }
          });
      assert false;
    } catch (IllegalArgumentException e) {
      assertEquals(
          "cannot use non-static local Creator to create actors; make it static (e.g. local to a static method) or top-level",
          e.getMessage());
    }
```

That won't fail, because it is using java assertions. Assertions aren't enabled by default. 

So, the `Props.create` pass (while the test tries to prove that it should fail). 

Then, the `assert` isn't executed and the `assertEquals` inside the catch block never gets executed.

Replace the `assert` by `org.junit.Assert.fail("above code should have failed");` and that test fails.

**Comment by octonato** [comment]
@Captain1653, do you want to take a stab on it and try to fix that test? By fixing, I mean, ensure that it will indeed fail the call and throw an `IllegalArgumentException`?

**Comment by Captain1653** [comment]
@ochrons Thank you for your explanation. I didn't guess))). Yes, I can try to do it. Should it be some different issue and different PR? For me - yes, because it's the different problem. What do you think? 

P.S. I tried to search places with "assert " in source code. It was used many times. 
Снимок экрана 2021-05-03 в 14 30 13

Is it worth to fix it in the different issue and different PR? (replace on assertEquals or similar)

Kind regards

**Comment by octonato** [comment]
Yes, you are right. I think we should fix it on a different PR. Let's not mix things.

**Review by octonato — APPROVED** [review]

**Review by johanandren — APPROVED** [review]

**Pull request merged**